### DISCUSSION TITLE
Add host process results in incorrect libvirtd settings for Ubuntu VERSION="24.04.1 LTS (Noble Numbat)"

### DISCUSSION THREAD

**Discussion by ffslcschools** [apache/cloudstack#9864] [Q&A] [answered]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME

~~~
Host agent processing script.
~~~

##### CLOUDSTACK VERSION

~~~
CloudStack 4.19.1.1
~~~

##### CONFIGURATION

Advanced networking

##### OS / ENVIRONMENT

PRETTY_NAME="Ubuntu 24.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.1 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo

##### SUMMARY

When adding a kvm host to Cloudstack, the agent configuration automatically adds the option libvirtd_opts='-l' to /etc/default/libvirtd. This creates failures for live migrations between cluster hosts. Documentation here https://docs.cloudstack.apache.org/en/latest/installguide/hypervisor/kvm.html is correct, stating that Ubuntu 20.04 and newer should use the option LIBVIRTD_ARGS="--listen".
The documentation is confusing in that it seems to tell you to make these changes manually, but the host provisioning process modifies all of the files referenced.

##### STEPS TO REPRODUCE

Simply adding a host via the web interface will do this at least when the host os version is Ubuntu 24.04.1 LTS.

~~~

~~~

##### EXPECTED RESULTS

~~~
The proper existing settings to be uncommented or at least added to the end of the config files as the documentation states.
~~~

##### ACTUAL RESULTS

Agent configuration process automatically adds the option libvirtd_opts='-l' to /etc/default/libvirtd when the os version is newer than 20.04.

~~~

~~~

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by weizhouapache** [comment]
If network/bridges have already been configured, ACS will not configure the host again.

The automatic configuration on kvm host is out of date, it does not work on some new distributions, due to libvirtd/netplan/nmcli etc.
Users are suggested to configure libvirtd, linux bridge or ovs manually before adding the host to cloudstack.

**Comment by ffslcschools** [comment]
Is there a way to disable the automatic configuration during host provisioning?

**Comment by weizhouapache** [comment]
> Is there a way to disable the automatic configuration during host provisioning?

@ffslcschools 
yes
if the network (linux bridges) have been configured correctly, the node is considered as preconfigured, ACS will not reconfigure it

**Comment by bwcxyk** [comment, marked as answer]
After using LIBVIRTD_ARGS=‘--listen’ option on Ubuntu22, libvirtd does not start. After running `systemctl mask libvirtd.socket libvirtd-ro.socket libvirtd-admin.socket libvirtd-tls.socket libvirtd-tcp.socket` libvirtd can be started.

**Reply by weizhouapache** [reply]
Good !
